src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/KestrelThread.cs

```diff
@@ -85,6 +97,22 @@ public void AllowStop()
 
         public void Stop(TimeSpan timeout)
         {
+            // Close and wait for all connections
+            ConnectionManager.WalkConnectionsAndClose();
+
+            // REVIEW: Should we use the timeout?
```

Comment:
Timing out WalkConnectionsAndClose seems fine, not sure about the disposable of UvWriteReq